### PR TITLE
[KYUUBI #5475][FOLLOWUP] Authz check permanent view's subquery should check view's correct privilege

### LINKED ISSUE #5475: [Subtask] Authz check permanent view's subquery should check view's correct privilege
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Authz check permanent view's subquery should check view's correct privilege

### Parent issue

#5474 

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To fix #5475 

In issue #5417 we fixed the problem that AUTHZ will still check scalar-subquery/in-subquery in permanent will.
But we just ignore the check, the subquery still will run, in this PR,  we record the permanent view's visited column to check the permanent view's privilege to avoid extra execution effort.

For the test `[KYUUBI #5417] should not check scalar-subquery in permanent view` I print all the plan that pass to privilege builder as below
截屏2023-10-19 下午4 05 46

before this pr
截屏2023-10-19 下午4 15 29

This two graph shows this pr deny the execution of subquery when we don't have  the veiw's privilege

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping  @yaooqinn

**Comment by cxzl25** [comment]
Incomplete image link in PR description.

**Comment by AngersZhuuuu** [comment]
> Incomplete image link in PR description.

Yea, changed

**Comment by AngersZhuuuu** [comment]
ping @yaooqinn @bowenliang123 could you table a look?

**Comment by AngersZhuuuu** [comment]
Any suggestion?

**Comment by yaooqinn** [comment]
Can we add some tests for it?

**Comment by AngersZhuuuu** [comment]
> Can we add some tests for it?

Existing ut already verified this? Since we skip check the subquery expression's privilege check

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala [hunk 1]
```diff
@@ -42,9 +42,16 @@ class RuleApplyPermanentViewMarker extends Rule[LogicalPlan] {
             // TODO: Currently, we do not do an auth check in the subquery
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala:42]: Comments overdue？

**Comment by yaooqinn** [comment]
The existing UT seems insufficient.

Let's update the test at least with `create view v1 as select id, name, max(age) as max_age from xxxx....` to test

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala]: Done

**Comment by AngersZhuuuu** [comment]
> The existing UT seems insufficient as it only has one col.
> 
> Let's update the test at least with `create view v1 as select id, name, max(age) as max_age from xxxx....` to test

How about current?

**Review by bowenliang123 — APPROVED** [review]
LGTM.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala [hunk 2]
```diff
@@ -39,12 +39,16 @@ class RuleApplyPermanentViewMarker extends Rule[LogicalPlan] {
...
-            subquery.withNewPlan(plan = PermanentViewMarker(subquery.plan, null))
+            subquery.withNewPlan(plan =
+              PermanentViewMarker(
+                subquery.plan,
+                permanentView.desc,
+                permanentView.output.map(_.name)))
         }
-        PermanentViewMarker(resolvedSubquery, resolvedSubquery.desc)
+        PermanentViewMarker(
+          resolvedSubquery,
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala:49]: nit: it's not resolvedSubquery but a view?

**Comment by yaooqinn** [comment]
Thanks, merged to master
